.github/workflows/go-check.yml

```
@@ -1,67 +1,18 @@
-# File managed by web3-bot. DO NOT EDIT.
-# See https://github.com/protocol/.github/ for details.
-
-on: [push, pull_request]
 name: Go Checks
 
+on:
+  pull_request:
+  push:
+    branches: ["master","release-v018","release-v022","release-v023","release-v026","release-v027","release-v028","release-v029"]
```

How did these end up here? This is just a convention, but we might do something different in the future. It also means we'd have to update workflow files for every release.

---

This is to stop running the workflows twice on pull requests. We can use our branch protection rule pattern here. For @web3-bot to be able to retrieve it, we'd need a higher level access (`gh api graphql -f query='query($owner: String!, $repo: String!) { repository(owner: $owner, name: $repo) { branchProtectionRules(first: 100) { nodes { pattern } } } }' -f owner=libp2p -f repo=go-libp2p --jq '.data.repository.branchProtectionRules.nodes'`).

We can update it once to:

```suggestion
    branches: ["master","release-v0[0-9][0-9]"]
```

---

> This is to stop running the workflows twice on pull requests.

Wasn't there a subtle difference between push and pull_request? One of them creates a fake merge commit, doesn't it?

---

> > This is to stop running the workflows twice on pull requests.
> 
> Wasn't there a subtle difference between push and pull_request? One of them creates a fake merge commit, doesn't it?

Yes, there is. `pull_request` creates a fake merge commit to the base branch of the PR (so it tests the "what if I merged this PR right now"), and `push` checks out the head branch as is.

To maintain the old behaviour we could get rid of the branches filter altogether:

```suggestion
```

After the updates to Unified CI I'm rolling out now, @web3-bot will start respecting changes like this in future rollouts, so we don't have to worry about it being overridden in the future.

The reason why I decided to add the branches filter in the first place is that it's a saner default for all repositories but diverging from that default is definitely OK.